woocommerce/woocommerce-ios
# Products Onboarding: Add visual UI for banner to My Store dashboard

Closes: #7976, #7977

## Description

This PR builds on the changes from https://github.com/woocommerce/woocommerce-ios/pull/7948 that introduced the UI for an announcement card on the My Store dashboard. That PR added support for displaying Just In Time Messages (JITMs) there.

To encourage new merchants to add their first product in the app, we will check if a store is eligible for products onboarding (if their store has 0 products) and then display the onboarding banner. As discussed in pe5pgL-11B-p2#comment-1080, this banner will take precedence over any JITMs shown in the same space on the dashboard.

## Changes

The most significant changes are in `DashboardViewModel.syncAnnouncements()`. This method is what we use to check if there is anything to display in the announcement card space, and set the appropriate view model as needed. We now have two checks:

* Check if a store is eligible for products onboarding and set that banner view model.
* If no onboarding banner was set, check if there are any JITMs for the store and set the relevant JITM view model.

Other changes:

* Moves the tracking event for products onboarding eligibility into the view model. Note: This event is **not** behind the feature flag. We are currently gathering these analytics and only want the UI to be behind a feature flag. To make sure this event is triggered regardless of feature flags, I've also removed the redundant `hasAnnouncementFeatureFlag` check from `DashboardViewController` — that flag is already checked within `DashboardViewModel.syncAnnouncements()`.
* Updates the `ProductStore` action we use to check eligibility for products onboarding, to make it easier to understand and use.

## Testing

On a store with no products:

1. Build and run the app in debug/alpha mode.
2. Confirm the products onboarding banner is shown.

Note that the dismiss and CTA buttons are not yet functional (changes to be made in #7978).

On a store with products:

1. Set a breakpoint in `DashboardViewModel` at L108 (where `self.syncJustInTimeMessages(for: siteID)` is called).
2. Build and run the app in debug/alpha mode.
4. Confirm the products onboarding banner is **not** shown.
5. Confirm the breakpoint is hit. (If your site is eligible for the test JITM, as described in pdfdoF-1uc-p2#comment-2581, you can continue and should see the JITM.)



## Screenshots

<img src="https://user-images.githubusercontent.com/8658164/198695461-d188cc4c-82aa-44b1-b5cd-ae9ef7f89709.png" width="300px">

## Submitter Checklist

Update release notes:
